custom_components/tahoma/climate_devices/somfy_heating_temperature_interface.py

```diff
@@ -0,0 +1,155 @@
+"""Support for Somfy Heating Temperature Interface."""
+import logging
+from typing import Optional
+
+from homeassistant.components.climate import SUPPORT_PRESET_MODE, ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_HEAT,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+OVERKIZ_TO_PRESET_MODES = {
+    OverkizCommandParam.SECURED: PRESET_AWAY,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.FREE: PRESET_NONE,
+}
+
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+
+OVERKIZ_TO_HVAC_MODES = {
+    OverkizCommandParam.AUTO: HVAC_MODE_AUTO,
+    OverkizCommandParam.MANU: HVAC_MODE_HEAT_COOL,
+}
+
+HVAC_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODES.items()}
+
+OVERKIZ_TO_HVAC_ACTION = {
+    OverkizCommandParam.COOLING: CURRENT_HVAC_COOL,
+    OverkizCommandParam.HEATING: CURRENT_HVAC_HEAT,
+}
+
+MAP_PRESET_TEMPERATURES = {
+    PRESET_COMFORT: OverkizState.CORE_COMFORT_ROOM_TEMPERATURE,
+    PRESET_ECO: OverkizState.CORE_ECO_ROOM_TEMPERATURE,
+    PRESET_AWAY: OverkizState.CORE_SECURED_POSITION_TEMPERATURE,
+}
+
+MODE_COMMAND_MAPPING = {
+    OverkizCommandParam.COMFORT: OverkizCommand.SET_COMFORT_TEMPERATURE,
+    OverkizCommandParam.ECO: OverkizCommand.SET_ECO_TEMPERATURE,
+    OverkizCommandParam.SECURED: OverkizCommand.SET_SECURED_POSITION_TEMPERATURE,
+}
+
+
+class SomfyHeatingTemperatureInterface(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Heating Temperature Interface."""
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    _attr_supported_features = SUPPORT_PRESET_MODE
+
+    def __init__(self, device_url: str, coordinator: OverkizDataUpdateCoordinator):
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
```

Comment:
So it works well?

Reply:
It's working, but linked_device shouldn't be an @property. 

The way to get the temperature also feel unnatural, it's different from the other devices. (no executor in the device), otherwise it's perfectly working.

Reply:
Good point for the property, they cannot take an argument.

About your feeling, it's normal. But we want to migrate all the other climate devices to this logic way more simple.